cmd/webhook/app/testing/testwebhook.go

```diff
@@ -56,7 +56,7 @@ type ServerOptions struct {
 	CAPEM []byte
 }
 
-func StartWebhookServer(t *testing.T, ctx context.Context, args []string, argumentsForNewServerWithOptions ...app.ServerOption) (ServerOptions, StopFunc) {
+func StartWebhookServer(t *testing.T, ctx context.Context, args []string, argumentsForNewServerWithOptions ...webhook.ServerOption) (ServerOptions, StopFunc) {
```

Comment:
Hm this is a good point - I'd made this one internal mostly so that `pkg/webhook` is a complete project-agnostic package for building webhooks (we now don't depend on any cert-manager APIs from there).

I didn't realise we were doing this, but I could move `internal/webhook` to `pkg/certmanagerwebhook` or something to resolve this?

Reply:
Tbh I quite like these things living under `internal`. Could `type ServerOption func(*server.Server)` not just be moved to `"github.com/jetstack/cert-manager/pkg/webhook/server"`?

Reply:
It could, however there's not actually a `NewServer` function in `pkg/webhook/server` at the moment to accept it. It's only accepted by the `internal/webhook.NewCertManagerWebhookServer` function at the moment (and that was a recent addition to allow @wallrj to override the ConversionHandler used during integration testing)